server/sonar-server/src/main/java/org/sonar/server/qualityprofile/ws/SearchAction.java

```diff
@@ -91,31 +111,61 @@ public void handle(Request request, Response response) throws Exception {
   }
 
   private static SearchWsRequest toSearchWsRequest(Request request) {
-    return  new SearchWsRequest()
+    return new SearchWsRequest()
+      .setOrganizationKey(request.param(PARAM_ORGANIZATION))
       .setProjectKey(request.param(PARAM_PROJECT_KEY))
       .setProfileName(request.param(PARAM_PROFILE_NAME))
       .setDefaults(request.paramAsBoolean(PARAM_DEFAULTS))
       .setLanguage(request.param(PARAM_LANGUAGE));
   }
 
-  private SearchWsResponse doHandle(SearchWsRequest request) {
-    SearchData data = dataLoader.load(request);
+  @VisibleForTesting
+  SearchWsResponse doHandle(SearchWsRequest request) {
+    validateRequest(request);
+    SearchData data = load(request);
     return buildResponse(data);
   }
 
+  private SearchData load(SearchWsRequest request) {
+    try (DbSession dbSession = dbClient.openSession(false)) {
+      OrganizationDto organization = wsSupport.getOrganizationByKey(dbSession, request.getOrganizationKey());
+      return new SearchData()
+        .setOrganization(organization)
+        .setProfiles(dataLoader.findProfiles(dbSession, request, organization))
+        .setActiveRuleCountByProfileKey(activeRuleIndex.countAllByQualityProfileKey())
+        .setActiveDeprecatedRuleCountByProfileKey(activeRuleIndex.countAllDeprecatedByQualityProfileKey())
+        .setProjectCountByProfileKey(dbClient.qualityProfileDao().countProjectsByProfileKey(dbSession));
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Remove this use of "countProjectsByProfileKey"; it is deprecated. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)